### ISSUE TITLE
NJS build error: implicit declaration of function

### ISSUE THREAD

**Issue by gornostay25** [nginx/njs#465]
/root/njs/nginx/../external/njs_webcrypto_module.c: In function 'njs_ext_derive':
/root/njs/nginx/../external/njs_webcrypto_module.c:1548:20: error: implicit declaration of function 'EVP_PKEY_new_mac_key'; did you mean 'EVP_PKEY_CTX_get0_pkey'? [-Werror=implicit-function-declaration]
 1548 |             pkey = EVP_PKEY_new_mac_key(EVP_PKEY_HMAC, NULL, k, length);
      |                    ^~~~~~~~~~~~~~~~~~~~
      |                    EVP_PKEY_CTX_get0_pkey
/root/njs/nginx/../external/njs_webcrypto_module.c:1548:41: error: 'EVP_PKEY_HMAC' undeclared (first use in this function); did you mean 'EVP_PKEY_RSA2'?
 1548 |             pkey = EVP_PKEY_new_mac_key(EVP_PKEY_HMAC, NULL, k, length);
      |                                         ^~~~~~~~~~~~~
      |                                         EVP_PKEY_RSA2
/root/njs/nginx/../external/njs_webcrypto_module.c:1548:41: note: each undeclared identifier is reported only once for each function it appears in
/root/njs/nginx/../external/njs_webcrypto_module.c: In function 'njs_ext_import_key':
/root/njs/nginx/../external/njs_webcrypto_module.c:1859:37: error: 'EVP_PKEY_HMAC' undeclared (first use in this function); did you mean 'EVP_PKEY_RSA2'?
 1859 |         pkey = EVP_PKEY_new_mac_key(EVP_PKEY_HMAC, NULL, key_data.start,
      |                                     ^~~~~~~~~~~~~
      |                                     EVP_PKEY_RSA2
cc -c -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g -I/root/boringssl/include -Wimplicit-function-declaration -Wno-deprecated-declarations  -I src/core -I src/event -I src/event/modules -I src/event/quic -I src/os/unix -I /root/pcre-8.45 -I /root/boringssl/.openssl/include -I /root/zlib-1.2.11 -I objs -I src/http -I src/http/modules -I src/http/v2 -I src/http/v3 -I /root/ngx_brotli/deps/brotli/c/include -I /root/njs/nginx/../src -I /root/njs/nginx/../build -I src/stream -I /root/njs/nginx/../src -I /root/njs/nginx/../build \
	-o objs/addon/src/ngx_http_headers_more_headers_out.o \
	/root/headers-more-nginx-module/src/ngx_http_headers_more_headers_out.c
cc1: all warnings being treated as errors
make[1]: *** [objs/Makefile:2096: objs/addon/external/njs_webcrypto_module.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/root/nginx-1.21.6-quic'
make: *** [Makefile:10: build] Error 2

I try to use -Wimplicit-function-declaration but its not work((

**Comment by xeioex** [comment]
Thanks for the report, the problem looks like a duplicate of #455
